geometry/eolearn/geometry/transformations.py

```diff
@@ -187,6 +187,7 @@ def _preprocess_vector_data(
         if self.values_column is not None and self.values is not None:
             values = [self.values] if isinstance(self.values, (int, float)) else self.values
             vector_data = vector_data[vector_data[self.values_column].isin(values)]
+            vector_data[self.values_column] = vector_data[self.values_column].astype(self.raster_dtype)
```

Comment:
would this mutate the original values?

Reply:
It seems like it might, but then it might be an issue already before, because there is some augmentation of the vector data going on.

You know what, I think it makes more sense to actually make sure that the input data is what one would expect, so perhaps it does make sense to fix it elsewhere. 

I'll just close this MR.

Reply:
:+1: it's hard to handle wrong input data